include/net/ipv4_autoconf.h

```diff
@@ -0,0 +1,65 @@
+/** @file
+ * @brief IPv4 Autoconfiguration
+ */
+
+/*
+ * Copyright (c) 2017 Matthias Boesl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __IPV4_AUTOCONF_H
+#define __IPV4_AUTOCONF_H
+
+#include <misc/slist.h>
+#include <stdint.h>
+
+/** Current state of IPv4 Autoconfiguration */
+enum net_ipv4_autoconf_state {
+	NET_IPV4_AUTOCONF_INIT,
+	NET_IPV4_AUTOCONF_PROBE,
+	NET_IPV4_AUTOCONF_ANNOUNCE,
+	NET_IPV4_AUTOCONF_CONFIGURED,
+};
+
+/* initial random delay*/
+#define IPV4_AUTOCONF_PROBE_WAIT 1
+/* number of probe packets */
+#define IPV4_AUTOCONF_PROBE_NUM 3
+/* minimum delay till repeated probe */
+#define IPV4_AUTOCONF_PROBE_MIN 1
+/* maximum delay till repeated probe */
+#define IPV4_AUTOCONF_PROBE_MAX 2
+/* number of announcement packets */
+#define IPV4_AUTOCONF_ANNOUNCE_NUM 2
+/* time between announcement packets */
+#define IPV4_AUTOCONF_ANNOUNCE_INTERVAL 2
+/* max conflicts before rate limiting */
+#define IPV4_AUTOCONF_MAX_CONFLICTS 10
+/* delay between successive attempts */
+#define IPV4_AUTOCONF_RATE_LIMIT_INTERVAL 60
+/* minimum interval between defensive ARPs */
+#define IPV4_AUTOCONF_DEFEND_INTERVAL 10
+
+/**
+ *  @brief Start IPv4 autocofiguration RFC 3927: IPv4 Link Local
+ *
+ *  @details Start IPv4 IP autoconfiguration
+ *
+ *  @param iface A valid pointer on an interface
+ */
+void net_ipv4_autoconf_start(struct net_if *iface);
+
+enum net_verdict net_ipv4_autoconf_input(struct net_if *iface,
+					 struct net_pkt *pkt);
```

Comment:
Indent looks off-by-one in this line (params should be aligned), could also be the github viewer issue.